boardswarm/src/qcomlt_debug_board.rs

```diff
@@ -0,0 +1,321 @@
+    // Power actuator
+    // TODO: The registered name is the serial port name rather than the node ?!?!?!
+    let power_actuator = QCOMLTDebugBoardActuator::new(port, "power".to_string());
+    r.register(&syspath, seqnum, power_actuator, properties);
+
+    /*
+    TODO:
+
+    2025-02-06T14:29:25.710019Z  INFO start: boardswarm::qcomlt_debug_board: Reserving console /dev/ttyACM0 for QCOMLT Debug Board 75eaaac075a445150202022343a460ff
+    2025-02-06T14:29:25.710237Z  INFO setup_subdevices{node="/dev/ttyACM0" seqnum=3}: boardswarm::qcomlt_debug_board: Setting up QCOMLTDebugBoard actuators for /dev/ttyACM0
+
+    2025-02-06T14:29:25.713803Z  INFO setup_subdevices{node="/dev/ttyACM0" seqnum=3}: boardswarm::qcomlt_debug_board: Ignoring outdated registration (seqnum: 3)
+    2025-02-06T14:29:25.713881Z  INFO setup_subdevices{node="/dev/ttyACM2" seqnum=1}: boardswarm::qcomlt_debug_board: Ignoring outdated registration (seqnum: 1)
+
+        */
+
+    // TODO: USB actuator
+    // TODO: Power button actuator
+    // TODO: Reset button actuator
+
+    // TODO: power measurement (voltage, current). The syntax is:
+    /*
+    0mV 0mA
+    [Pp]ower [Uu]sb [s]tatus pwr[Bb]tn [Rr]stbtn
+    */
+}
+
+// TODO add other debug board commands
+enum QCOMLTDebugBoardCommand {
+    PowerOn(tokio::sync::oneshot::Sender<Result<(), std::io::Error>>),
+    PowerOff(tokio::sync::oneshot::Sender<Result<(), std::io::Error>>),
+}
+
+async fn process(
+    mut port: tokio_serial::SerialStream,
+    mut exec: tokio::sync::mpsc::Receiver<QCOMLTDebugBoardCommand>,
+) {
+    while let Some(command) = exec.recv().await {
+        match command {
+            /* TODO: Create a standalone rust library for the QCOMLTDebugBoard
+            similar to https://github.com/boardswarm/mediatek-brom rather than
+            sending raw bytes here. */
+            QCOMLTDebugBoardCommand::PowerOn(sender) => {
+                info!("Sending PowerOn command");
+                let r = port.write_all(b"P\n").await;
+                let _ = sender.send(r);
+            }
+            QCOMLTDebugBoardCommand::PowerOff(sender) => {
+                info!("Sending PowerOff command");
+                let r = port.write_all(b"p\n").await;
+                let _ = sender.send(r);
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+struct QCOMLTDebugBoardActuator {
+    device: tokio::sync::mpsc::Sender<QCOMLTDebugBoardCommand>,
+    name: String,
+}
+
+impl QCOMLTDebugBoardActuator {
+    fn new(port: tokio_serial::SerialStream, name: String) -> Self {
+        let (device, exec) = tokio::sync::mpsc::channel(16);
+        tokio::spawn(process(port, exec));
+        Self { device, name }
+    }
+}
+
+/*
+TODO: calling change-mode in cli directly does not work (but setting mode does). Does it also not work for the other implementations ?
+
+obbardc@nox:~/projects/git/boardswarm$ RUST_LOG=info cargo run --bin boardswarm-cli -- -i localhost actuator 1 change-mode on
```

Comment:
actuators don't have a mode change; they take a json string to do their settings, similar to what you configure in the device section wrt. modes. This is to support differnet types of actuators especially ones that aren't just boolean :)

Reply:
OH, change-mode expects json format:

```
Change actuator mode

Usage: boardswarm-cli actuator <ACTUATOR> change-mode <MODE>

Arguments:
  <MODE>  Actuator specific mode in json format

Options:
  -h, --help  Print help
```

Reply:
that's what i said! `change-mode` might not be the best phrasing, but typically that's purely internal.. Conceptually the actuator only actuates as requested, it's only a device that knows what it needs for a given mode.

Fwiw atm it's a bit silly as we just have GPIO's; However for a controllable bench supply rather then just on/off you'd configure a few more parameters (e.g. voltage, over current protection etc).

Reply:
Right, but passing JSON in isn't clear and the response when you pass e.g. `on` instead of json string doesn't really match. So I will open an issue/PR to clarify that :-)

Reply:
fwiw for the record this was reported as !165